modules/core/src/v2/coins/trx.ts

```diff
@@ -137,13 +141,25 @@ export class Trx extends BaseCoin {
   }
 
   signTransaction(params: TronSignTransactionOptions): SignedTransaction {
-    const tx = params.txPrebuild.transaction;
+    const tx = params.txPrebuild.txInfo.transaction;
     // this prv should be hex-encoded
-    const halfSigned = tronWeb.utils.crypto.signTransaction(params.prv, tx);
-    if (_.isEmpty(halfSigned.signature)) {
+    const transaction = tronWeb.utils.crypto.signTransaction(params.prv, tx);
+    if (_.isEmpty(transaction.signature)) {
```

Comment:
it's possible this has a signature on it and is invalid. (so a fully signed tx coming through this flow would have 1 sig). this depends on previous state. we should check if it came in unsigned, does it have 1 sig? and then if it was half, does it have 2?

Reply:
I'm not sure how useful this is going to be since I would expect the builder to check for that, but I'll add it.